packages/ui-components/src/Input/TextTypeInput.tsx

```diff
@@ -0,0 +1,170 @@
+import React, {
+  ChangeEvent,
+  CSSProperties,
+  useCallback,
+  useEffect,
+  useMemo,
+  useState,
+  FunctionComponent,
+} from "react";
+import classNames from "classnames/bind";
+import styles from "./styles.module.scss";
+import { CommonInputProps, CommonInput } from "./BaseInput";
+import { generateContainerClassnames } from "../utils/objectToClassnames";
+import { isEmpty } from "lodash";
+
+export interface TextAndNumberProps<T = string | number> {
+  initialValue?: T;
+  value?: T;
+  maxLength?: number;
+  onChange?: (
+    event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>,
+  ) => void;
+  suffix?: JSX.Element;
+  onBlur?: (
+    event: React.FocusEvent<HTMLInputElement | HTMLTextAreaElement> | undefined,
+  ) => void;
+  onFocus?: (
+    event: React.FocusEvent<HTMLInputElement | HTMLTextAreaElement> | undefined,
+  ) => void;
+  multiline?: boolean;
+  placeholder?: string;
+  // prop only used internally for implementation
+  textType?: "text" | "email" | "password" | "number";
+  // this defines the input type rendered by this field
+  // for multiline text types, this will be <textarea>
+  // for text/email/password/number types this will be <input> with a type field
+  JSXInput?: JSX.Element;
+};
+
+// the following props are used to implement {Email, Password, Number}Input
+// it is recommended to use those components instead of passing an "email" type to TextInput
+interface CustomProps {
+  prefix?: JSX.Element;
+  ariaLabel?: string;
+  ariaLabelledBy?: string;
+  autoFocus?: boolean;
+  tabIndex?: number;
+  onKeyDown?: (
+    event: React.KeyboardEvent<HTMLInputElement>,
+  ) => void;
+  onClick?: (
+    event: React.MouseEvent<HTMLInputElement, MouseEvent>,
+  ) => void;
+  forgotPasswordLinkDiv?: JSX.Element;
```

Comment:
I don't think we should limit this element to a "div". Since it's type is a JSX element, perhaps we should call it `forgotPasswordLinkElement`?